app/services/pii/cacher.rb

```diff
@@ -19,9 +19,18 @@ def save(user_password, profile = user.active_profile)
     end
 
     def fetch
-      decrypted_pii = user_session[:decrypted_pii]
-      return unless decrypted_pii
-      Pii::Attributes.new_from_json(decrypted_pii)
+      pii_string = fetch_string
```

Comment:
Are you thinking of using this same tooling to handle the PII that is moving through the proofing flow? I'm curious what that looks like if so.

May be an irrelevant question in an FSMv2 world where we store the unverified PII on the client. But I suspect we'll want the changes to reduce KMS calls in place before we get there.

Reply:
Ideally, yeah. It kind of requires a more central way to manage which FSMv2 can hopefully bring.

We do know the top-level keys that are used (`idv/doc_auth` and `idv`), but one of those is probably accessed during each proofing step anyway? It may be easier to encrypt/decrypt those in their entirety during the proofing process.